plugins/outputs/splunkhec/splunkhec.go

```diff
@@ -0,0 +1,210 @@
+	Index  string      `json:"index,omitempty"`
+	Source string      `json:"source,omitempty"`
+	Host   string      `json:"host"`
+	Fields interface{} `json:"fields"`
+}
+
+func NewSplunkHEC() *SplunkHEC {
+	return &SplunkHEC{}
+}
+
+func (d *SplunkHEC) Connect() error {
+	if d.Token == "" {
+		return fmt.Errorf("token is a required field for Splunk HEC output")
+	}
+	if d.Url == "" {
+		return fmt.Errorf("url is a required field for Splunk HEC output")
+	}
+
+	d.client = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: d.Timeout.Duration,
+	}
+	return nil
+}
+
+func (d *SplunkHEC) Write(metrics []telegraf.Metric) error {
+
+	const redactedApiKey = "****************"
+
+	if len(metrics) == 0 {
+		return nil
+	}
+	var hecPostData string
+
+	for _, m := range metrics {
+		if hecMs, err := buildMetrics(m, d); err == nil {
+			hecPostData = hecPostData + string(hecMs)
+		} else {
+			log.Printf("I! unable to build Metric for %s, skipping\n", m.Name())
+		}
+	}
+
+	req, err := http.NewRequest("POST", d.Url, strings.NewReader(hecPostData))
+	if err != nil {
+		return fmt.Errorf("unable to create http.Request, %s\n", strings.Replace(err.Error(), d.Token, redactedApiKey, -1))
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", "Splunk "+d.Token)
+	// Add the Request-Channel header incase Indexer Acknowledgment is enabled.
+	req.Header.Add("X-Splunk-Request-Channel", d.Token)
+
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error POSTing metrics, %s\n", strings.Replace(err.Error(), d.Token, redactedApiKey, -1))
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode > 209 {
+		body, _ := ioutil.ReadAll(resp.Body)
+		return fmt.Errorf("received bad status code, %d [%s]\n", resp.StatusCode, body)
+	}
+
+	return nil
+}
+
+func (d *SplunkHEC) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *SplunkHEC) Description() string {
+	return "Configuration for Splunk HEC to send metrics to.\nDoes not make use of Indexer Acknowledgement"
+}
+
+func buildMetrics(m telegraf.Metric, d *SplunkHEC) (string, error) {
+
+	var metricGroup string
+	dataGroup := HECTimeSeries{}
+	obj := map[string]interface{}{}
+
+	for k, v := range m.Fields() {
+		// Empty out anything in obj since I don't know how golang scoping works...
+		for k, _ := range obj {
+			delete(obj, k)
+		}
+		dataGroup = HECTimeSeries{}
+		obj["metric_name"] = m.Name() + "." + k
+		obj["_value"] = v
+
+		dataGroup = HECTimeSeries{
+			Time:   float64(m.Time().UnixNano() / 1000000000),
+			Event:  "metric",
+			Fields: obj,
+		}
+
+		if d.Source != "" {
+			dataGroup.Source = d.Source
+		} else {
+			dataGroup.Source = "telegraf"
+		}
+		if d.Index != "" {
+			dataGroup.Index = d.Index
+		}
+
+		if !verifyValue(v) {
+			// Reset the dataGroup
+			dataGroup = HECTimeSeries{}
+			continue
+		}
+
+		// Need to get Host from m.Tags()
+		buildHecTags(m, &dataGroup)
```

Comment:
in this function, you override the Fields field of dataGroup.
is this really what you intend to do?